packages/skill-lesson/hooks/use-mux-player.tsx

```diff
@@ -63,14 +64,18 @@ export const VideoProvider: React.FC<
       })
     : null
 
-  const {data: abilityRules, status: abilityRulesStatus} =
-    trpcSkillLessons.modules.rules.useQuery({
-      moduleSlug: module.slug.current,
-      moduleType: module.moduleType,
-      lessonSlug: exerciseSlug,
-      sectionSlug: section?.slug,
-      isSolution: lesson._type === 'solution',
-    })
+  const {
+    data: abilityRules,
+    status: abilityRulesStatus,
+    refetch: refetchAbilityRules,
+    isRefetching,
```

Comment:
with full subscriber stored in a cookie we get to call `refetchAbilityRules` which reflects new abilities without having to refresh the page.